crates/network-scanner/src/ping.rs

```diff
@@ -0,0 +1,49 @@
+use std::net::Ipv4Addr;
+
+use network_scanner_net::tokio_raw_socket::TokioRawSocketStream;
+use network_scanner_proto::icmp_v4;
+
+use tokio::io::{AsyncReadExt, AsyncWriteExt};
+use tracing::trace;
+
+pub async fn ping(ip: Ipv4Addr) -> anyhow::Result<()> {
+    let mut socket = TokioRawSocketStream::connect(ip)
+        .await
+        .map_err(|e| anyhow::anyhow!(e))?;
```

Comment:
suggestion: With anyhow, you don’t need `map_err` in such cases.
```rust
let mut socket = TokioRawSocketStream::connect(ip).await?;
```
But you could use [`with_context`](https://docs.rs/anyhow/latest/anyhow/trait.Context.html#tymethod.with_context) to add more context to the error:
```rust
let mut socket = TokioRawSocketStream::connect(ip)
    .await
    .with_context(|| format!("couldn’t connect to {ip}"))?;
```

Reply:
updated